### PR TITLE
[Compressible Agent] [Tool Call] Compressible agent tool calls

### PR DESCRIPTION
**Pull request by yenif**
## Why are these changes needed?

Add tool support to compressible agent

* https://github.com/microsoft/autogen/pull/791
* https://github.com/microsoft/autogen/pull/974

## Checks

- [x] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [x] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [x] I've made sure all auto checks have passed.

### CODE REVIEW

**Comment by kevin666aa** [comment]
@yenif Please take a look at #1024. Can you create a PR that adds generate_tool_reply (modify only client.py and conversable_agent.py only), and add a test for that?

This is changing the core functionalities and we will get more people to review this.

**Comment by yenif** [comment]
@kevin666aa that's this one https://github.com/microsoft/autogen/pull/974. This pr builds on that PR, so it appears to also include all the same changes until #974 is merged. Normally I'd chain the PRs to diff correctly, but it seems that is not possible on cross fork PRs.

#1005 then builds on this PR, thus it currently shows all of the changes for all 3 PRs

**Comment by yenif** [comment]
I should be able to review this tonight for any updates now that converable tools are merged

File: autogen/agentchat/contrib/compressible_agent.py [hunk 1]
```diff
@@ -256,21 +261,28 @@ def _manage_history_on_token_limit(self, messages, token_used, max_token_allowed
...
             else:
                 raise ValueError(
                     "Message can't be converted into a valid ChatCompletion message. Either content or function_call must be provided."
                 )
-        if "function_call" in oai_message:
-            oai_message["role"] = "assistant"  # only messages with role 'assistant' can have a function call.
-            oai_message["function_call"] = dict(oai_message["function_call"])
+
+        if oai_message.get("function_call", False) or oai_message.get("tool_calls", False):
+            oai_message["role"] = "assistant"  # only messages with role 'assistant' can have a function or tool calls.
```
Comment (qingyun-wu) [inline autogen/agentchat/contrib/compressible_agent.py:280]: Are `oai_message["function_call"] = dict(oai_message["function_call"])` and `oai_message["tool_calls"] = dict(oai_message["tool_calls"])` needed?

**Comment by qingyun-wu** [comment]
Just have one question (might be minor) in my comment. Otherwise LGTM!

File: autogen/agentchat/contrib/compressible_agent.py (hunk 1 above)
Comment (yenif) [reply autogen/agentchat/contrib/compressible_agent.py:280]: https://github.com/microsoft/autogen/pull/974#discussion_r1426987232 comment suggested these are not needed any more

**Comment by sonichi** [comment]
I'm not sure if we want to keep maintaining compressible agent. We want to ultimately implement compression as one agent capability similar to teachability. There is discussion in #1073. @kevin666aa @qingyun-wu please indicate who's the current owner  and what's your suggested plan.

**Comment by yenif** [comment]
I've been thinking about switching to a different strategy, so abandoning CompressibleAgent works for me.

**Comment by ekzhu** [comment]
@yenif @sonichi @kevin666aa @qingyun-wu If everyone agrees we can close this PR for now.

**Pull request closed without merging**